delta-io/connectors
# Flink Delta Source PR_8 based on PR 9 - Partition support

PR Plan:
PR 5 - Option support in Bounded Mode
PR 6 - Continuous mode without monitoring for changes
PR 6.1 - Package Refactoring
PR 7 - Monitoring for changes in Continuous mode
PR 7.1 More tests for PR 7 including IT case tests for Continuous mode.
PR 9 - Builder
**PR 8 - Partition Support** <- PR 9 was done before PR 8, hence PR 8 is based on PR 9.
PR 10 - Get BATCH_SIZE for Format builder from Source options - this value is currently hardcoded.

-- All functionalities are in place up to this point. The following PRs are extra tests, eventual bug fixes and code/javadoc adjustments. --

PR 11 - Annotate with @deprecated  Sink's builder methods that names starts from "with". Add new ones without "with" prefix and overload partitions with List argument for Sink.
PR 12 - Add validation for option names and values including using not applicable options for given mode via generic option method.
PR 13 - Additional IT tests including all options and end to end tests using Source and Sink.

"Begin with the end in mind" -> aim to release first week or so of June, e.g. June 7
Priorities this week:
PR 8 - partitioning support -> Initial PR ready for May 5th.

While @kristoffSC is on vacation
@tdas + @scottsand-db to review public API
Once Krzysztof is back
README / docs
QA
end-to-end integration tests - using Sink to write, Source to read, etc.
more examples / migrate examples from internal flink module to examples folder

TODOs and other improvements planed as next:

- Use index for keeping track of already processed paths in SnapshotProcessor - needs new Delta API.
- Do do not create snapshot when streaming only changes (startingVersion or startingTimestamp option used). - needs new Delta API since for startingTimestamp we still need to create a snapshot. There is no API like deltaLog.getChangesFromTimestamp(....)
- Refactor Test util classes to make them common for sink and source.
- Re-visit Delta Standalone APIs for when the source see's metadata/protocol actions. Update ActionProcessor class to handle MetaData and Protocol Actions
